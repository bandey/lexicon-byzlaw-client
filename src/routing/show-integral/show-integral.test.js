import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

import enzymeToJson from 'enzyme-to-json';

import {BrowserRouter} from 'react-router-dom';
import DataProvider from '../../data-provider/data-provider.js';
import Lexicon from '../../components/lexicon/lexicon.js';

import ShowIntegral from './show-integral.js';

const data = [
  {w: 'one', c: 1},
  {w: 'two', c: 2},
  {w: 'three', c: 3},
];

// Module.mock must be in the same scope as import Module
jest.mock('../../data-provider/data-provider.js');
DataProvider.mockImplementation(({children, query}) => {
  const Children = children;
  return (<Children data={data} />);
});

jest.mock('../../components/lexicon/lexicon.js');
Lexicon.mockImplementation(({data, WrapLink}) => {
  return (
    <React.Fragment>
      {data.map(item => WrapLink({item, children: item.c}))}
    </React.Fragment>
  );
});

it('uses correct query', () => {
  const rendered = shallow(<ShowIntegral match={{params: {lingua: 'lat'}}} />);

  // console.log(rendered.debug());

  const provider = rendered.find('DataProvider');
  expect(provider.exists()).toBeTruthy();
  expect(provider.prop('query')).toEqual({
    name: 'lexiconIntegral',
    gql: 'lexiconIntegral(lingua: "lat") { w c }',
  });
});

it('renders correct content', () => {
  const rendered = mount(
    <BrowserRouter>
      <ShowIntegral match={{params: {lingua: 'lat'}, url: '/route/integral/lat'}} />
    </BrowserRouter>
  );

  // console.log(rendered.debug());

  expect(enzymeToJson(rendered)).toMatchSnapshot();
});
