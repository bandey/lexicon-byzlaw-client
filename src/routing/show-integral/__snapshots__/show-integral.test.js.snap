// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct content 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ShowIntegral
      match={
        Object {
          "params": Object {
            "lingua": "lat",
          },
          "url": "/route/integral/lat",
        }
      }
    >
      <DataProvider
        query={
          Object {
            "gql": "lexiconIntegral(lingua: \\"lat\\") { w c }",
            "name": "lexiconIntegral",
          }
        }
      >
        <AdapterChoice
          data={
            Array [
              Object {
                "c": 1,
                "w": "one",
              },
              Object {
                "c": 2,
                "w": "two",
              },
              Object {
                "c": 3,
                "w": "three",
              },
            ]
          }
        >
          <Lexicon
            WrapLink={[Function]}
            data={
              Array [
                Object {
                  "c": 1,
                  "w": "one",
                },
                Object {
                  "c": 2,
                  "w": "two",
                },
                Object {
                  "c": 3,
                  "w": "three",
                },
              ]
            }
          >
            <Link
              key="one"
              to="/route/integral/lat/one"
            >
              <LinkAnchor
                href="/route/integral/lat/one"
                navigate={[Function]}
              >
                <a
                  href="/route/integral/lat/one"
                  onClick={[Function]}
                >
                  1
                </a>
              </LinkAnchor>
            </Link>
            <Link
              key="two"
              to="/route/integral/lat/two"
            >
              <LinkAnchor
                href="/route/integral/lat/two"
                navigate={[Function]}
              >
                <a
                  href="/route/integral/lat/two"
                  onClick={[Function]}
                >
                  2
                </a>
              </LinkAnchor>
            </Link>
            <Link
              key="three"
              to="/route/integral/lat/three"
            >
              <LinkAnchor
                href="/route/integral/lat/three"
                navigate={[Function]}
              >
                <a
                  href="/route/integral/lat/three"
                  onClick={[Function]}
                >
                  3
                </a>
              </LinkAnchor>
            </Link>
          </Lexicon>
        </AdapterChoice>
      </DataProvider>
    </ShowIntegral>
  </Router>
</BrowserRouter>
`;
